actions/global_actions.jsx

```diff
@@ -313,15 +313,21 @@ export async function redirectUserToDefaultTeam() {
         }
 
         let channelName = LocalStorageStore.getPreviousChannelName(user.id, team.id);
-        const channel = getChannelByName(state, channelName);
+        let channel = getChannelByName(state, channelName);
+        const channelMember = getMyChannelMember(state, channelName);
 
-        if (channel && channel.team_id === team.id) {
+        if (channel && channel.team_id === team.id && channelMember) {
             dispatch(selectChannel(channel.id));
             channelName = channel.name;
         } else {
             const {data} = await dispatch(getChannelByNameAndTeamName(team.name, channelName));
```

Comment:
Why do we need to invoke `getChannelByName` and if that doesn't work invoke `getChannelByNameAndTeamName`?

Reply:
My understanding is that `getChannelByName` is the selector (gets the info from the state) and if it fails, `getChannelByNameAndTeamName` tries to fetch the result from the server.

Reply:
Oh right, my mistake I didn't notice one was a selector and the other an action. Thanks.